wp-includes/components/class-wp-service-worker-offline-commenting-component.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * WP_Service_Worker_Offline_Commenting_Component class.
+ *
+ * @package PWA
+ */
+
+/**
+ * Class representing the service worker core component for handling offline commenting.
+ *
+ * @since 0.2
+ */
+class WP_Service_Worker_Offline_Commenting_Component implements WP_Service_Worker_Component {
+
+	/**
+	 * Internal storage for replacements to make in the offline commenting handling script.
+	 *
+	 * @since 0.2
+	 * @var array
+	 */
+	protected $replacements = array();
+
+	/**
+	 * Adds the component functionality to the service worker.
+	 *
+	 * @since 0.2
+	 *
+	 * @param WP_Service_Worker_Scripts $scripts Instance to register service worker behavior with.
+	 */
+	public function serve( WP_Service_Worker_Scripts $scripts ) {
+
+		$scripts->register(
+			'wp-offline-commenting',
+			array(
+				'src'  => array( $this, 'get_script' ),
+				'deps' => array( 'wp-base-config', 'wp-error-response' ),
+			)
+		);
+
+		if ( ! is_admin() ) {
+			$offline_template_url = add_query_arg( 'wp_error_template', 'offline', home_url( '/' ) );
```

Comment:
@todo These originally have filtering

Reply:
Probably there should be one method for getting the `offline_template_url` instead of duplicating from the error response handling logic.